plugins/lua/zone.lua

```diff
@@ -1227,18 +1227,21 @@ local function check_valid_unit()
         and dfhack.units.isFortControlled(unit)
         and dfhack.units.isAlive(unit)
         and dfhack.units.isAnimal(unit)
+        or dfhack.units.isTamable(unit)  -- account for caged animals
+        and unit.flags1.caged
+        and not unit.flags1.tame -- dont show on merchant animals
 end
 
 local function is_geldable()
     local unit = dfhack.gui.getSelectedUnit(true)
 
-    return unit and dfhack.units.isGeldable(unit)
+    return unit and dfhack.units.isGeldable(unit) and dfhack.units.isFortControlled(unit) and not unit.flags3.gelded
```

Comment:
```suggestion
    return unit and dfhack.units.isGeldable(unit) and dfhack.units.isFortControlled(unit) and not dfhack.units.isGelded(unit)
```

Reply:
this is a more complicated check than just the flag -- should use the API function